pamelalozano/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,93 @@
 
 package com.google.sps;
 
+import java.util.HashSet;
+import java.util.logging.Logger;
+import java.util.LinkedList;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ListIterator;
+
 import java.util.Collection;
 
 public final class FindMeetingQuery {
+    
+  private static Logger log = Logger.getLogger(FindMeetingQuery.class.getName());
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    //No atendees
+    if (request.getAttendees().size() <= 0 && request.getOptionalAttendees().size() <= 0) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    } 
+
+    //Exceedes time
+    if(TimeRange.WHOLE_DAY.duration() < request.getDuration()){
+        return Arrays.asList();
+    }
+
+    List<Event> eventsList = new ArrayList<>(events);
+    HashMap<String, List<TimeRange>> eventsMap = new HashMap<>();
```

Comment:
You might want to check out `MultiMap` from Guava. Makes this kind of thing much easier.

https://guava.dev/releases/19.0/api/docs/com/google/common/collect/Multimap.html

Reply:
https://github.com/google/guava/wiki/NewCollectionTypesExplained#multimap